slackminion/bot.py

```diff
@@ -174,7 +174,7 @@ async def stop(self):
             self.webserver.stop()
         self.plugin_manager.unload_all()
 
-    def send_message(self, channel, text, thread=None, reply_broadcast=None, attachments=None, parse=None,
+    async def send_message(self, channel, text, thread=None, reply_broadcast=None, attachments=None, parse=None,
```

Comment:
we can't just change these methods to `async` as they are being called by synchronous plugin functions which will break.

Reply:
What has been done in the past when `async` was added to a function such as for get_channels (in 204a75812a3f9363d0f8c125b9dc3ae32d366505)?

My plan was to upgrade the plugins before importing this new change but for that to work having the typing information available for async/await helps a lot etc.

I am open to other suggestions.